src/ptvsd/server/api.py

```diff
@@ -68,12 +106,44 @@ def debug(address, log_dir=None, multiprocess=True):
 
 @_starts_debugging
 def enable_attach(dont_trace_start_patterns, dont_trace_end_patterns):
-    server_opts.host, server_opts.port = pydevd._enable_attach(
-        (server_opts.host, server_opts.port),
+    host, port = pydevd._enable_attach(
+        ("127.0.0.1", 0),
         dont_trace_start_patterns=dont_trace_start_patterns,
         dont_trace_end_paterns=dont_trace_end_patterns,
         patch_multiprocessing=server_opts.multiprocess,
     )
+
+    log.info("pydevd debug server running at: {0}:{1}", host, port)
+    _DAPMessagesListener.initialize()
+
+    with pydevd.skip_subprocess_arg_patch():
+        import subprocess
+        adapter_args = [
+            sys.executable,
+            os.path.join(os.path.dirname(ptvsd.__file__), "adapter"),
+            "--host",
+            server_opts.host,
+            "--port",
+            str(server_opts.port),
+            "--connect-to-port",
+            str(port)
+        ]
+
+        if common_opts.log_dir is not None:
+            adapter_args += ["--log-dir", common_opts.log_dir]
+
+        log.info(
+            "enable_attach() spawning attach-to-PID debugger injector: {0!r}", adapter_args
+        )
+
+        # Adapter life time is expected to be longer than this process,
+        # so never wait on the adapter process
+        subprocess.Popen(adapter_args, bufsize=0)
```

Comment:
If the adapter life is expected to be longer, the `PyDevJsonCommandProcessor.on_terminate_request` shouldn't kill that process when terminate is called -- as right now it'll kill all child processes by default.

i.e.: we may need to add an api to add pids which shouldn't be killed by `pydevd` when terminating and use it here (although this is probably a different task).

Reply:
Created a task for it: https://github.com/microsoft/ptvsd/issues/1786